vanilla-framework/vanilla-framework
# Pattern nesting

## Done

Unnested sass used solely to generate classnames.

e.g.:
```
.foo {
  color: #990000;

  &__bar {
    padding: 0;
  }
}
```
becomes:
```
.foo {
  color: #990000;
}

.foo__bar {
  padding: 0;
}
```

This was a decision taken following discussion in a developer catch up. Key benefits include:
- Full classnames are now searchable in files
- Nesting depths are reduced
- Generated CSS is closer to the source SASS, allowing more mindful CSS generation
  

Also fixed some class names that weren't proper BEM.

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- See no visual differences

## Details

Fixes #2091
